lib/syskit/models/configured_deployment.rb

```diff
@@ -18,10 +18,14 @@ class ConfiguredDeployment
             # @return [Hash] the name mappings, e.g. the mapping from a task
             #   name in {#model} to the name this task should have while running
             attr_reader :name_mappings
+            # @return [Array] the tasks related to this deployment supposed to be set as
+            #   read only
+            attr_reader :read_only
 
+            # rubocop:disable Metrics/ParameterLists
             def initialize(
                 process_server_name, model, name_mappings = {},
-                process_name = model.name, spawn_options = {}
+                process_name = model.name, spawn_options = {}, read_only = []
```

Comment:
I think it would be better having `read_only` as a keyword argument (same for the other methods you modified in e.g. the tests).